### PR TITLE
Implement settings API on admin/settings/microservices & admin/settings/analytics

### LINKED ISSUE #527: Implement settings API on admin/settings/microservices and admin/settings/analytics
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-frontend

image
image

### PR DESCRIPTION
**Pull request by CosmicCoder96**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

- Implement settings API on admin/settings/microservices
- Implement settings API on admin/settings/analytics

Fixes #527

### CODE REVIEW

**Comment by CosmicCoder96** [comment]
Deployment link : 
http://cc96-opev-fe-delta.herokuapp.com/admin/settings/microservices

**Comment by CosmicCoder96** [comment]
@niranjan94 @geekyd Please review

File: app/templates/components/forms/admin/settings/microservices-form.hbs [hunk 1]
```diff
@@ -6,7 +6,7 @@
     <label>
       {{t 'API Server'}}
     </label>
-    {{input type='url' name='android_app' placeholder='http://droidgen.eventyay.com/api/v2/'}}
+    {{input type='url' name='android_app' placeholder='http://droidgen.eventyay.com/api/v2/' value=settings.androidAppUrl}}
   </div>
```
Comment (niranjan94) [inline app/templates/components/forms/admin/settings/microservices-form.hbs]: Use url input widget (component)

**Comment by CosmicCoder96** [comment]
@geekyd @niranjan94 Please review

**Comment by CosmicCoder96** [comment]
@niranjan94 Please review/merge this.

File: app/components/forms/admin/settings/analytics-form.js [hunk 2]
```diff
@@ -5,6 +5,7 @@ const { Component } = Ember;
 export default Component.extend({
   actions: {
     submit() {
```
Comment (niranjan94) [inline app/components/forms/admin/settings/analytics-form.js]: Do we need this action ?

Comment (niranjan94) [inline app/components/forms/admin/settings/analytics-form.js]: Can you directly make use of `save` ?

**Comment by niranjan94** [comment]
And why have all validations been removed ?

**Comment by CosmicCoder96** [comment]
@niranjan94 The only validations were that the fields had a valid url type, since we are using the widget it includes the http part itself, so removed the validation.

**Comment by niranjan94** [comment]
@CosmicCoder96 the widget takes care of just `http` and `https`. The rest should still be validated. 

For example, `http://some random line which is not a valid domain name` is an invalid url.

**Comment by CosmicCoder96** [comment]
@niranjan94 I have tried to use a regular expression which should allow urls of the form 
`images.google.com` and `droidgen.eventyay.com/v1/v2` can you please review thanks !

**Comment by niranjan94** [comment]
Microservices can also be an IP address. Can also have ports in them. All
of these are valid URIs

**Comment by CosmicCoder96** [comment]
@niranjan94 using this https://gist.github.com/dperini/729294

File: app/components/forms/admin/settings/microservices-form.js [hunk 3]
```diff
@@ -15,7 +15,8 @@ export default Component.extend(FormMixin, {
           identifier : 'android_app',
           rules      : [
             {
-              type   : 'url',
+              type   : 'regExp',
+              value  : /^(?:\S+(?::\S*)?@)?(?:(?!(?:10|127)(?:\.\d{1,3}){3})(?!(?:169\.254|192\.168)(?:\.\d{1,3}){2})(?!172\.(?:1[6-9]|2\d|3[0-1])(?:\.\d{1,3}){2})(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)(?:\.(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)*(?:\.(?:[a-z\u00a1-\uffff]{2,}))\.?)(?::\d{2,5})?(?:[/?#]\S*)?$/i,
```
Comment (CosmicCoder96) [inline app/components/forms/admin/settings/microservices-form.js]: @niranjan94 This should include the ip addresses and ports, I tested it with my own IP and some fake port, seemed to work. I have modified it slightly from the gist version to remove the http/https validation.

Comment (niranjan94) [inline app/components/forms/admin/settings/microservices-form.js]: Add this to `utils/validators.js` and reference from there so that it can be re-used.

Comment (CosmicCoder96) [inline app/components/forms/admin/settings/microservices-form.js]: @niranjan94 Should this replace the exisitng url validation in the validator?

Comment (niranjan94) [inline app/components/forms/admin/settings/microservices-form.js]: No.

**Comment by CosmicCoder96** [comment]
@niranjan94 I have added it.

File: app/utils/validators.js [hunk 4]
```diff
@@ -11,6 +11,40 @@ const validUrlPattern = new RegExp('^(https?:\\/\\/)?' // protocol
   + '(\\?[;&a-z\\d%_.~+=-]*)?' // query string
   + '(\\#[-a-z\\d_]*)?$', 'i'); // fragment locator
 
+export const noProtocolValidUrlPattern = new RegExp(
```
Comment (niranjan94) [inline app/utils/validators.js]: Call it `protocolLessValidUrlPattern`

**Comment by CosmicCoder96** [comment]
@niranjan94 Done

**Comment by CosmicCoder96** [comment]
@niranjan94 please review

**Pull request merged**